components/contact/show_inquiry_modal.coffee

```diff
@@ -41,12 +49,37 @@ module.exports = class ShowInquiryModal extends ContactView
     return unless city = @partner.displayLocations @user?.get('location')?.city
     @$('.contact-location').html ", " + city
 
-  submit: (e) ->
+  onSubmit: (e) ->
+    super
+
     analyticsHooks.trigger 'inquiry:show',
       label: modelNameAndIdToLabel 'show', @show.get('id')
+
     @model.set
       inquireable_id: @show.get('id')
       inquireable_type: 'partner_show'
       contact_gallery: true
       session_id: SESSION_ID
-    super
+
+    if sd.FORCED_LOGIN_INQUIRY is 'force_login' and !@user
```

Comment:
This is the magic part of this, we had to split submit into two methods, `onSubmit` is the click callback, `submit` actually sends the inquiry. This way we can intercept here and open up the flow to ensure that a user is either logged in or registered before an inquiry gets made.